app/src/main/java/ca/ualbert/cs/tasko/UserProfileActivity.java

```diff
@@ -44,16 +57,47 @@ protected void onCreate(Bundle savedInstanceState) {
         username = (EditText)findViewById(R.id.UserProfileActivityUsername);
         phoneNumber = (EditText)findViewById(R.id.UserProfileActivityPhoneNumber);
         emailAddress = (EditText)findViewById(R.id.UserProfileActivityEmail);
+
+        confirmButton = (Button) findViewById(R.id.UserProfileConfirmButton);
+        cancelButton = (Button) findViewById(R.id.UserProfileCancelButton);
         User user = CurrentUser.getInstance().getCurrentUser();
 
         if(CurrentUser.getInstance().loggedIn()){
 
             emailAddress.setText(user.getEmail());
-            username.setText(user.getUsername());
+            username.setText(user.getName());
             phoneNumber.setText(user.getPhoneNumber());
             Log.i("User logged in", emailAddress.getText().toString() );
-
         }
 
+        ConfirmButtonClick();
+        CancelButtonClick();
+    }
+
+    public void ConfirmButtonClick(){
+        confirmButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+
+                setResult(RESULT_OK);
+                CurrentUser.getInstance().getCurrentUser().setName(username.getText().toString());
```

Comment:
One thing I would suggest is that if you are going to use the current user singleton multiple times maybe make it a variable so you don't have to write it out each time.